packages/cli/test/unit/Telemetry.test.ts

```diff
@@ -281,23 +281,34 @@ describe('Telemetry', () => {
 			pulseSpy.mockClear();
 		});
 
-		xtest('should trigger pulse in intervals', () => {
+		test('should trigger pulse in intervals', async () => {
 			expect(pulseSpy).toBeCalledTimes(0);
 
 			jest.advanceTimersToNextTimer();
+			await Promise.resolve();
```

Comment:
maybe we could extract this into a helper, and use `setImmediate` for resolving the promises, to give any pending promises some time to finish?
something like this
```typescript
const flushPromises = async () => new Promise((resolve) => setImmediate(resolve));
```

```typescript
await flushPromises();
```

Reply:
`const flushPromises = async () => new Promise(setImmediate);` and `const flushPromises = () => new Promise(setImmediate);` trigger timeouts (not sure why), but `export const flushPromises = () => Promise.resolve(setImmediate);` works fine so went with that.

Reply:
`const flushPromises = () => Promise.resolve(setImmediate);` creates a new resolved promise with `setImmediate` as the value. it's not actually using the `setImmediate` to let any dangling promises in the current event-loop from finishing.

Reply:
Got it, I see now, I'll dig into these timeouts then. Also something about this setup is OOMing the CI.

Reply:
@freyamade Adi might have traced this back to the PH SDK:

> btw, the timeout issues are actually expected. we most certainly have handling promises that aren't being resolved. the flushPromise thing will only work if all promises resolve in a timely manner.
adding a simple await Promise.resolve() isn't going to help with promises that never resolve :confused: (edited) 
in this case, looks like the dangling promise is from the posthog sdk.

Up to you if we want to spend more time to unskip these two tests at this stage.

Reply:
Related #4930